uaclient/files/notices.py

```diff
@@ -0,0 +1,153 @@
+    ):
+        if not override_dir:
+            self.directory = "/var/lib/ubuntu-advantage/notices/"
+            if not is_permanent:
+                self.directory = "/var/run/notices/"
+            self.override = False
+        else:
+            self.directory = override_dir
+            self.override = True
+        self.is_root = root_mode
+        self.is_permanent = is_permanent
+
+    def add(self, notice_label_id: NoticeFileDetails, description: str):
+        label = notice_label_id.label
+        file_order_id = notice_label_id.order_id
+        if self.is_root:
+            notice_files = self.get_all_notice_files()
+            labels_present = [file.split("-")[-1] for file in notice_files]
+            if label in labels_present:
+                return
+            filename = "{}-{}".format(file_order_id, label)
+            system.write_file(
+                os.path.join(self.directory, filename), description
+            )
+        else:
+            raise exceptions.NonRootUserError
+
+    def try_add(self, notice_label_id: NoticeFileDetails, description: str):
+        try:
+            self.add(notice_label_id, description)
+        except exceptions.NonRootUserError:
+            event.warning("Trying to add notice as non-root user")
+
+    def remove(self, notice_label_id: NoticeFileDetails):
+        label = notice_label_id.label
+        file_order_id = notice_label_id.order_id
+        filename = "{}-{}".format(file_order_id, label)
+        if self.is_root:
+            notice_files = self.get_all_notice_files()
+            label_file = [file for file in notice_files if file == filename]
+            if len(label_file) == 1:
+                system.remove_file(os.path.join(self.directory, label_file[0]))
+        else:
+            raise exceptions.NonRootUserError
+
+    def try_remove(self, notice_label_id: NoticeFileDetails):
+        try:
+            self.remove(notice_label_id)
+        except exceptions.NonRootUserError:
+            event.warning("Trying to add notice as non-root user")
+
+    def get_all_notice_files(self):
+        if not os.path.exists(self.directory):
+            os.makedirs(self.directory)
+
+        notices = [
+            file
+            for file in os.listdir(self.directory)
+            if os.path.isfile(os.path.join(self.directory, file))
+        ]
+        notices.sort()
+        return notices
+
+    def read(self):
+        if not self.override:
+            notice_directories = (
+                "/var/lib/ubuntu-advantage/notices/",
+                "/var/run/notices/",
+            )
+        else:
+            notice_directories = (self.directory,)
+        notices = []
+        for notice_directory in notice_directories:
+            notices = [
+                file
+                for file in os.listdir(notice_directory)
+                if os.path.isfile(os.path.join(notice_directory, file))
+            ]
+            notices.sort()
```

Comment:
Can we use `get_all_notice_files` somehow here ? We would need to pass the `notice_directory` as a parameter, but maybe it is worth it

Reply:
Ah, it could work.